Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java

```diff
@@ -95,4 +107,21 @@
         
         return new ResponseEntity<String>(msg.getXML(xmlEntries), HttpStatus.OK);
     }
+
+    private void addAlternativeIds(String id, ConceptEntry entry)
+            throws IllegalAccessException, LuceneException, IndexerRunningException {
+
+        // Check if this id is same as the id sent by user
+        if (!entry.getId().equalsIgnoreCase(id)) {
+            entry.getAlternativeIds().add(id);
+        }
+
+        // Fetch the internal concept id based on the wordnetid
+        Map<String, String> fieldMap = new HashMap<String, String>();
+        fieldMap.put(SearchFieldNames.WORDNETID, entry.getWordnetId());
+        ConceptEntry[] conceptEntries = indexService.searchForConcepts(fieldMap, "AND");
```

Comment:
Hm, I think there is still an issue here. We have three different ids, right?:
- Conceptpower ID (e.g. http://www.digitalhps.org/concepts/ba626314-5d54-41b6-8f41-0013be5269be)
- Specific Wordnet ID (e.g. WID-02612368-V-01-occur)
- General Wordnet ID (e.g. WID-02612368-V-??-occur) -> I believe it just takes the first word when given an id with '??' in it.

In your code you simply search for a concept with the given Wordnet id and add the id of the concept. But when I search for example for the specific wordnet id, then alternative ids has only the CP id and that's it. but would it should have is 
- id of concept returned is CP id if such an id exists, otherwise specific WordNet Id.
- alternative ids should have CP id (for simplicity), specific WordNet id, and if the general WordNet id was used to request a concept, then the general WordNet Id (otherwise I think you can omit the general WordNet ID). 

I don't think you need to use Lucene here at all. You can simply get the concept by its ID and get all information from the concept. Only thing to keep in mind is that you need to retrieve the concept again if the ID used in the request was not the one used to create a concept wrapper (e.g. for 'use' ```WID-02612368-V-??-occur``` was used instead of ```WID-02612368-V-01-occur```, but for other wrappers the specific Wordnet Id was used).

Does this make sense?